hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -126,31 +129,41 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     boolean acquiredLock = false;
     IOException exception = null;
     OMClientResponse omClientResponse = null;
+    List<OmKeyInfo> missingParentInfos = new ArrayList<>();
+
     try {
       // check Acl
       checkKeyAcls(ozoneManager, volumeName, bucketName, keyName,
           IAccessAuthorizer.ACLType.CREATE, OzoneObj.ResourceType.KEY);
 
       // Check if this is the root of the filesystem.
       if (keyName.length() == 0) {
-        return new OMDirectoryCreateResponse(null,
+        return new OMDirectoryCreateResponse(null, null,
             omResponse.setCreateDirectoryResponse(
-                CreateDirectoryResponse.newBuilder()).build());
+                CreateDirectoryResponse.newBuilder()).build(), false);
       }
       // acquire lock
       acquiredLock = omMetadataManager.getLock().acquireWriteLock(BUCKET_LOCK,
           volumeName, bucketName);
 
       validateBucketAndVolume(omMetadataManager, volumeName, bucketName);
 
+      Path keyPath = Paths.get(keyName);
+
       // Need to check if any files exist in the given path, if they exist we
       // cannot create a directory with the given key.
       OMFileRequest.OMDirectoryResult omDirectoryResult =
           OMFileRequest.verifyFilesInPath(omMetadataManager,
-          volumeName, bucketName, keyName, Paths.get(keyName));
+          volumeName, bucketName, keyName, keyPath);
+
+      List<String> missingParents = OMFileRequest.getMissingParents(
+          omMetadataManager, volumeName, bucketName, keyPath.getParent());
 
       OmBucketInfo omBucketInfo = omMetadataManager.getBucketTable().get(
           omMetadataManager.getBucketKey(volumeName, bucketName));
+      long baseObjId = OMFileRequest.getObjIdFromTxId(transactionLogIndex);
+      long objectCount = 1;
+
       OmKeyInfo dirKeyInfo = null;
       if (omDirectoryResult == FILE_EXISTS ||
```

Comment:
I believe this can be moved up to after the "verifyFilesInPath" call. Also, should we repeat the presence of intermediate directory path in keyTable in getMissingParents() again?

Reply:
will move the code snippet as suggested.
would there be any benefit checking for the presence of intermediate directory in keyTable again? the objective of the function, as the name suggests is focussed on getting parents in the path which are missing.

Reply:
I think we can combine the two functions to avoid doing DB reads twice. We could probably change OMDirectoryResult structure to include the result + missing dirs.

Reply:
Yes, @hanishakoneru's suggestion makes sense if we want to avoid doing the same reads twice.

Reply:
Thanks, makes sense. Will make the suggested change.